derp/derphttp/derphttp_client.go

```diff
@@ -185,6 +185,18 @@ func (c *Client) useHTTPS() bool {
 	if c.url != nil && c.url.Scheme == "http" {
 		return false
 	}
+	// Turn on development mode and connect to DERP using plain HTTP.
+	// Questions:
+	// * Should this be a command line option or an envknob?
+	// * If envknob, should it be stubbed out for iOS, etc.?
```

Comment:
Not sure we need to worry about iOS on a per-knob basis. #3707 is open about allowing envknobs to be configured on platforms where it's harder to set environment variables, when we get around to that this knob will also become settable there.

Reply:
Thanks! I was wondering in particular if this should be added to magicsock/debugknobs{,stubs}.go since there is another envknob in this file that isn't in there, and I'm not clear on how we decide what goes in there and what doesn't?

Reply:
@bradfitz can say for sure, but I think that file is for magicsock-specific knobs, and this is operating at a slightly different level. 

We also used to be a lot more memory constrained on iOS (before iOS 15, https://tailscale.com/blog/go-linker/ has the full story), which would perhaps explain why we didn't want to include `envknob` on iOS, but we're OK with it nowadays.